### ISSUE TITLE
error "OpenCV 4.x+ requires enabled C++11 support"

### ISSUE THREAD

**Issue by kyuhyong** [Tossy0423/yolov4-for-darknet_ros#17]
Before apt update and upgrade recently, the package worked.
Now I am getting following errors when tried catkin_make again.

```
In file included from /usr/include/opencv4/opencv2/core/types_c.h:82:0,
                 from /home/r1mini/catkin_ws/src/yolov4-for-darknet_ros/darknet_ros/darknet_ros/include/darknet_ros/image_interface.h:13,
                 from /home/r1mini/catkin_ws/src/yolov4-for-darknet_ros/darknet_ros/darknet_ros/src/image_interface.c:9:
/usr/include/opencv4/opencv2/core/cvdef.h:690:4: error: #error "OpenCV 4.x+ requires enabled C++11 support"
 #  error "OpenCV 4.x+ requires enabled C++11 support"
    ^~~~~
compilation terminated due to -Wfatal-errors.
yolov4-for-darknet_ros/darknet_ros/darknet_ros/CMakeFiles/darknet_ros_lib.dir/build.make:2282: recipe for target 'yolov4-for-darknet_ros/darknet_ros/darknet_ros/CMakeFiles/darknet_ros_lib.dir/src/image_interface.c.o' failed
make[2]: *** [yolov4-for-darknet_ros/darknet_ros/darknet_ros/CMakeFiles/darknet_ros_lib.dir/src/image_interface.c.o] Error 1
```
I have build installed opencv 3.4 on jetson nano using below cmake option
```
$ cmake -D CMAKE_BUILD_TYPE=RELEASE \
-D CMAKE_INSTALL_PREFIX=/usr/local \
-D ENABLE_CXX11=ON \
-D WITH_TBB=OFF \
-D WITH_IPP=OFF \
-D WITH_1394=OFF \
-D BUILD_WITH_DEBUG_INFO=OFF \
-D BUILD_DOCS=OFF \
-D INSTALL_C_EXAMPLES=ON \
-D INSTALL_PYTHON_EXAMPLES=ON \
-D BUILD_EXAMPLES=OFF \
-D BUILD_TESTS=OFF \
-D BUILD_PERF_TESTS=OFF \
-D WITH_QT=OFF \
-D WITH_GTK=ON \
-D WITH_OPENGL=ON \
-D OPENCV_EXTRA_MODULES_PATH=../../opencv_contrib-3.4.0/modules \
-D WITH_V4L=ON  \
-D WITH_FFMPEG=ON \
-D WITH_XINE=ON \
-D BUILD_NEW_PYTHON_SUPPORT=ON \
-D PYTHON2_INCLUDE_DIR=/usr/include/python2.7 \
-D PYTHON2_NUMPY_INCLUDE_DIRS=/usr/lib/python2.7/dist-packages/numpy/core/include/ \
-D PYTHON2_PACKAGES_PATH=/usr/lib/python2.7/dist-packages \
-D PYTHON2_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython2.7.so \
-D PYTHON3_INCLUDE_DIR=/usr/include/python3.6m \
-D PYTHON3_NUMPY_INCLUDE_DIRS=/usr/lib/python3/dist-packages/numpy/core/include/  \
-D PYTHON3_PACKAGES_PATH=/usr/lib/python3/dist-packages \
-D PYTHON3_LIBRARY=/usr/lib/x86_64-linux-gnu/libpython3.6m.so \
../
```

I also tried to add below option in CMakeList.txt but no luck.
set(CMAKE_CXX_STANDARD 11)
set(CMAKE_CXX_STANDARD_REQUIRED TRUE)

Any suggestions?
Thanks!

**Comment by Theiby** [comment]
Dİd you find any solutions ?

**Comment by kyuhyong** [comment]
It actually turned out that the issue was caused by the cv_bridge not being built with C++ 11 support. 
The only way I could solve this issue was uninstalling cv_bridge and manually build it from source.
When removing cv_bridge, it will also uninstall many other packages so I had to use dpkg -r --force-depends to remove just one package.
One draw back is that once cv_bridge is removed, apt upgrade might not work correctly.
I don't know if cv_bridge will resolve this issue anytime soon.
